src/InstrumentationEngine.Lib/refcount.cpp

```diff
@@ -129,82 +125,76 @@ namespace MicrosoftInstrumentationEngine
 
     void CRefCount::RecordCreation(CRefCount* pCRefCount)
     {
-        if (g_fRecordingInitialized)
-        {
-            EnterCriticalSection(&g_crst);
+        EnterCriticalSection(&g_crst);
 
-            CRefRBMap::CPair* pPair = g_map.Lookup(pCRefCount);
-            if (pPair)
-            {
-                //VSASSERT(false, "Object recorder: Object destroyed without the recorder being notified.");
-                TRACE(_T("Object \"%s\" (%lxh) destroyed without recorder being notified."), pCRefCount->m_pszCRefCountClassName, pCRefCount);
-            }
+        CRefRBMap::CPair* pPair = g_map.Lookup(pCRefCount);
+        if (pPair)
+        {
+            //VSASSERT(false, "Object recorder: Object destroyed without the recorder being notified.");
+            TRACE(_T("Object \"%s\" (%lxh) destroyed without recorder being notified."), pCRefCount->m_pszCRefCountClassName, pCRefCount);
+        }
 
-            g_map.SetAt(pCRefCount, 1);
+        g_map.SetAt(pCRefCount, 1);
 
-            LeaveCriticalSection(&g_crst);
-        }
+        LeaveCriticalSection(&g_crst);
     }
 
     void CRefCount::RecordDestruction(CRefCount* pCRefCount)
     {
-        if (g_fRecordingInitialized)
+        if (pCRefCount->m_ulcRef <= 0)
```

Comment:
Did we actually hit this condition? That suggests a pretty bad bug.

Reply:
I don't know. It was probably in there when it was first written. 

I'll fold this back into the caller so that it is more similar to how it was before, but this check us used everywhere the CModuleRefCount pattern is used.